### PR TITLE
Here is the updated integration of the DDI

### PR DESCRIPTION
**Pull request by juggernautsei**
The new integration of the DDI. This is still the free version. Would like some feedback on how to move forward with this and would is this a feature that should have a global to turn it on.

### CODE REVIEW

**Comment by juggernautsei** [comment]
@sjpadgett can you give this a once over?

File: controllers/C_Prescription.class.php [hunk 1]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
         $this->pconfig = $GLOBALS['oer_config']['prescriptions'];
         $this->RxList = new RxList();
 
+        $interaction = "";
+
+        /*
+        *   check to see if RxNorm installed
+        */
+        $rxn = sqlQuery("SELECT table_name FROM information_schema.tables WHERE table_name = 'RXNCONSO' OR table_name = 'rxnconso'");
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: you should be able to do this instead:
$rxn = sqlQuery("SELECT table_name FROM information_schema.tables WHERE table_name = ?", array(mitigateSqlTableUpperCase('RXNCONSO'));

File: controllers/C_Prescription.class.php [hunk 2]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
...
+            }
+            if (count($nameList) < 2) {
+                $interaction = "Need more than one drug.";
+
+            } else {
+                /*
+                *  If there are drugs to compare, collect the data
+                */
+                $rxcui_list = implode(", ", $nameList);
+                $data = file_get_contents("https://rxnav.nlm.nih.gov/REST/interaction/list.json?rxcuis=".$rxcui_list);
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: use function to encode parameters in the url (stops special chars from breaking anything):
```
$data = file_get_contents("https://rxnav.nlm.nih.gov/REST/interaction/list.json?rxcuis=" . urlencode($rxcui_list));
```

File: controllers/C_Prescription.class.php [hunk 3]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
...
         $this->RxList = new RxList();
 
+        $interaction = "";
+
+        /*
+        *   check to see if RxNorm installed
+        */
+        $rxn = sqlQuery("SELECT table_name FROM information_schema.tables WHERE table_name = 'RXNCONSO' OR table_name = 'rxnconso'");
+        if ($rxn == false) {
+           $interaction = "Could not find RxNorm Table! Please install.";
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: translate above with just the xl() function

File: controllers/C_Prescription.class.php [hunk 4]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
...
+           $pid = $GLOBALS['pid'];
+            $medList = sqlStatement("SELECT drug FROM prescriptions WHERE active = 1 AND patient_id = ?", array($pid));
+            $nameList = array();
+            while ($name = sqlFetchArray($medList)) {
+                $drug = explode(" ", $name['drug']);
+                $rXn = sqlQuery("SELECT `rxcui` FROM `" . mitigateSqlTableUpperCase('RXNCONSO') . "` WHERE `str` LIKE ?", array("%" . $drug[0] . "%"));
+                $nameList[] = $rXn['rxcui'];
+            }
+            if (count($nameList) < 2) {
+                $interaction = "Need more than one drug.";
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: translate above with just the xl() function

File: controllers/C_Prescription.class.php [hunk 5]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
...
+                if (!empty($json['fullInteractionTypeGroup'][0]['fullInteractionType'])) {
+                    foreach ($json['fullInteractionTypeGroup'][0]['fullInteractionType'] as $item) {
+                        $interaction = xlt('Comment').":".text($item['comment'])."</br>";
+                        $interaction .= xlt('Drug1 Name{{Drug1 Interaction}}').":".text($item['minConcept'][0]['name'])."</br>";
+                        $interaction .= xlt('Drug2 Name{{Drug2 Interaction}}').":".text($item['minConcept'][1]['name'])."</br>";
+                        $interaction .= xlt('Severity').":". text($item['interactionPair'][0]['severity'])."</br>";
+                        $interaction .= xlt('Description').":". text($item['interactionPair'][0]['description']);
+                    }
+                } else {
+                    $interaction = 'No interactions found';
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: translate above with just the xl() function

File: templates/prescription/general_list.html [hunk 6]
```diff
@@ -135,7 +135,14 @@
 
 {if $prescriptions}
 <span class="title"><b>{xl t='List'|escape:'html'}</b></span>
-
+<div id="drug-drug">
+    <h3>Drug-Drug Interaction</h3>
```
Comment (bradymiller) [inline templates/prescription/general_list.html]: translate (and escape):
```
<h3>{xl t='Drug-Drug Interaction'|text}</h3>
```

File: templates/prescription/general_list.html [hunk 7]
```diff
@@ -135,7 +135,14 @@
...
+    <div id="return_info">
+        {$INTERACTION}
```
Comment (bradymiller) [inline templates/prescription/general_list.html]: escape:
```
{$INTERACTION|text}
```

File: templates/prescription/general_list.html [hunk 8]
```diff
@@ -135,7 +135,14 @@
...
+    </div>
+    <p title="Severity information will be missing if interaction is found"><a href="#" style="font-size: x-small">*Notice</a></p>
```
Comment (bradymiller) [inline templates/prescription/general_list.html]: translate (and escape):
```
<p title="{xl t='Severity information will be missing if interaction is found'|attr}"><a href="#" style="font-size: x-small">*{xl t='Notice'|text}</a></p>
```

**Comment by bradymiller** [comment]
Placed some comments (just my normal translation/escaping stuff :) )

**Comment by juggernautsei** [comment]
@bradymiller do you think the page needs a loading spinner?

**Comment by bradymiller** [comment]
Looks like you have a php syntax error:
https://travis-ci.org/openemr/openemr/jobs/466479959

File: controllers/C_Prescription.class.php [hunk 9]
```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
         $this->pconfig = $GLOBALS['oer_config']['prescriptions'];
         $this->RxList = new RxList();
 
+        $interaction = "";
+
+        /*
+        *   check to see if RxNorm installed
+        */
+        $rxn = sqlQuery("SELECT table_name FROM information_schema.tables WHERE table_name = ?", array(mitigateSqlTableUpperCase('RXNCONSO'));
```
Comment (bradymiller) [inline controllers/C_Prescription.class.php]: looks like need another right parenthesis at end to fix the php syntax error

**Comment by sjpadgett** [comment]
Are drug interaction a part of M.U?

**Comment by bradymiller** [comment]
Continued this here:
https://github.com/openemr/openemr/pull/2065
(so will close this PR)

**Pull request closed without merging**